packages/runtime/plugin-testing/src/cli/index.ts

```diff
@@ -26,6 +26,26 @@ export const mergeUserJestConfig = (testUtils: TestConfigOperator) => {
   }
 };
 
+export const getJestTransformEsModulesRegStr = () => {
+  const esmModulesInPnpm = [
+    '@modern-js\\+runtime@',
+    '@modern-js\\+plugin@',
+    // @modern-js-reduck+store, @modern-js-reduck+effects and so on
+    '@modern-js-reduck',
+    '@babel\\+runtime@',
+  ];
+  // yarn or npm
+  const esmModules = [
+    '@modern-js/runtime',
+    '@modern-js/plugin',
+    '@modern-js-reduck',
+    '@babel/runtime',
+  ];
+  return `node_modules/(?!(\\.pnpm/(${esmModulesInPnpm.join(
```

Comment:
Is `.pnpm` required? Maybe remove it can work well in pnpm projects

Reply:
Pnpm will create two layers of `node_modules`, such as: `node_modules/.pnpm/@modern-js+runtime@0.0.0/node_modules/@modern-js/runtime/package.json`

So remove the `\\.pnpm/` should work.

Reply:
I removed `\\.pnpm/` and the test case didn't pass. Can you give me an example?

Reply:
> It will not match `'node_modules/.pnpm/@modern-js+runtime@0.0.0'`.

I made a mistake here, when `\\.pnpm/` is removed, it matches to `'node_modules/.pnpm/@modern-js+runtime@0.0.0'`, but the result we want is a mismatch. Only if it doesn't match will jest process the package.

Reply:
The rule here is to match all paths that do not contain `node_modules/.pnpm/@modern-js+runtime@`

Reply:
ok, I get it👌